k8s/src/bin/server.rs

```diff
@@ -0,0 +1,229 @@
+use persia_operator::crd::PersiaJobSpec;
+use persia_operator::PersiaJobResources;
+
+use actix_web::{get, post, web, App, HttpRequest, HttpServer, Responder};
+use kube::client::Client;
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+
+static KUBERNETES_CLIENT: once_cell::sync::OnceCell<Client> = once_cell::sync::OnceCell::new();
+
+#[derive(Deserialize, Serialize)]
+struct JobIdentifier {
+    pub job_name: String,
+    pub namespace: String,
+}
+
+#[derive(Deserialize, Serialize)]
+struct PodIdentifier {
+    pub pod_name: String,
+    pub namespace: String,
+}
+
+#[derive(Deserialize, Serialize)]
+struct ApplyRequest {
+    pub job_identifier: JobIdentifier,
+    pub spec: PersiaJobSpec,
+}
+
+#[derive(Deserialize, Serialize)]
+struct ExecutionResults {
+    pub success: bool,
+    pub err_msg: Option<String>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct ListResponse {
+    pub execution_results: ExecutionResults,
+    pub resources: Option<Vec<String>>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct PodsResponse {
+    pub execution_results: ExecutionResults,
+    pub body: Option<String>,
+}
+
+#[post("/apply")]
+async fn apply(req: web::Json<ApplyRequest>) -> impl Responder {
+    let kubernetes_client = KUBERNETES_CLIENT
+        .get()
+        .expect("KUBERNETES_CLIENT not set")
+        .clone();
+    let resources = PersiaJobResources::new(
+        &req.spec,
+        &req.job_identifier.job_name,
+        &req.job_identifier.namespace,
+        kubernetes_client,
+    );
+    let resp = match resources.apply().await {
+        Ok(_) => ExecutionResults {
+            success: true,
+            err_msg: None,
+        },
+        Err(e) => ExecutionResults {
+            success: false,
+            err_msg: Some(format!("{:?}", e)),
```

Comment:
`to_string` is simple than `format`

Reply:
Fixed in https://github.com/PersiaML/PERSIA/pull/142/commits/4b7070554666fd7c94ef897a0160e2031c1ab7f0